dataprocspawner/spawner.py

```diff
@@ -302,18 +306,31 @@ async def poll(self):
 
 ################################################################################
 # User form functions
-################################################################################
+################################################################################ 
   def _options_form_default(self):
     """ Builds form using values passed by administrator either in Terraform
     or in the jupyterhub_config_tpl.py file.
     """
+    config_files = ""
+    file_extension = ".yaml"
     
-    # Skips the form if no config provided.
-    if not self.dataproc_configs:
+    if self.dataproc_configs.endswith(file_extension):
```

Comment:
Do we already require that configs end with .yaml? I think right now the config can end with anything, and I think it doesn't necessarily make sense to enforce that they need to end with .yaml.

One alternative is to try list_gcs_files first, and we if we get a non-empty/non-error result we can assume that this is a directory. Otherwise treat it as a file. Does that sound reasonable?

Reply:
Actually thinking about this more, do we want to support cases like having a list of directories, or having a list where some entries are individual config files and some are directories?

Reply:
Admins can pass any file type but for now the code only [parses yaml](https://github.com/GoogleCloudDataproc/jupyterhub-dataprocspawner/blob/master/dataprocspawner/spawner.py#L428).

I can definitely add better checks to differentiate files from bucket/folders.

Regarding a list of directories and files, I am not sure if we want to allow admins to do that. It does bring flexibility but having config files all over the place might complicate lifecycle. Seems like we should channel admins to have one single place for all of them.

Happy to implement it though. I see value in both options and flexibility is always a plus. We anyway have to keep the file list for backward compatibility.

Reply:
I agree that it's probably confusing to munge folders and files in a single list.

Let's stick with the current logic, where you can have either a single folder or a list of files.